Rework owners_file.py and add custom exception

mgoerens opened the pull request:
closes #317

komish commented:
I would prefer changes of this size wait until after we finish rolling out the helm dependency bugfix.

mgoerens commented:
@komish now that the helm dependency bugfix is merged, we could look into getting this merged. I've rebased and fixed merge conflicts. Note that the work has been separated into two commits for better readability / reviewability.

komish commented on scripts/src/owners/owners_file.py:
```diff
@@ -8,70 +9,67 @@
     path = os.path.join("charts", category, organization, chart, "OWNERS")
-    status, owner_content = get_owner_data_from_file(path)
-    return status, owner_content
+    success = True
+
+    try:
+        owner_content = get_owner_data_from_file(path)
+    except OwnersFileError as e:
+        print(f"Error getting OWNERS file data: {e}")
+        success = False
+
+    return success, owner_content
```
Nit: Can we assert success instead of asserting failure?
```suggestion
    success = False

    try:
        owner_content = get_owner_data_from_file(path)
        success = True
    except OwnersFileError as e:
        print(f"Error getting OWNERS file data: {e}")

    return success, owner_content
```

komish commented on scripts/src/owners/owners_file.py:
```diff
@@ -8,70 +9,67 @@
-    except Exception:
-        pass
-    return vendor
+    vendor_name = ""
+    with contextlib.suppress(KeyError):
+        vendor_name = owner_data["vendor"]["name"]
+    return vendor_name
```
I was originally going to suggest this instead (to avoid importing `contextlib`):

```python
return owner_data.get("vendor", {}).get("name", "")
```

But, actually, my implementation (and yours) both suffer from a failure if owner_data is not a dictionary. AttributeError would get thrown for mine (calling the get method), and TypeError for yours (NoneType is not subscriptable).

Better to leave this as is?

---

[EDIT]: I'm seeing now that there are several places where this logic was being simplified, but we'll run into the problem cases because we can't assert the owner_data input's type.

komish reviewed (changes requested):
Couple of small things to consider. My apologies for the delay.

komish reviewed (approved):
/lgtm
